src/cirrus/lib/statedb.py

```diff
@@ -139,7 +141,11 @@ def get_items_page(self, collections_workflow: str,
             startkey = { key: dbitem[key] for key in ['collections_workflow', 'itemids', 'state_updated', 'updated']}
             resp = self.query(collections_workflow, state=state, since=since, sort_ascending=sort_ascending, sort_index=sort_index, Limit=limit, ExclusiveStartKey=startkey, )
         else:
-            resp = self.query(collections_workflow, state=state, since=since,  sort_ascending=sort_ascending, sort_index=sort_index, Limit=limit)
+            if error_begins_with:
```

Comment:
Seems like this won't work with paging (i.e., when needing to fetch more than 300 records), as the query above the `else` using the `ExclusiveStartKey` is what is used for paging.

Reply:
I think with your suggested kwargs approach it will work now. I'm testing manually locally currently. I've left the commented-out individual calls to `query` simply for review purposes, temporarily.